### PR TITLE
e2e test for search logs

### PR DESCRIPTION
**Pull request by baurine**

### CODE REVIEW

**Comment by baurine** [comment]
What did:

- e2e test for search logs
- use expect-puppeteer to simplify the code, not rely on ant design specific selectors as much as possible

hi @breeswish , please help take a look, thanks!

File: ui/tests/e2e/search_log.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      const searchForm = await page.waitForSelector('form[name="search_form"]')
+
+      // set log level to INFO
+      await ppExpect(searchForm).toClick('div#log_level_selector')
+      const logLevelLists = await page.waitForSelector('ul[role=listbox]')
+      await ppExpect(logLevelLists).toClick('li', { text: 'INFO' })
+
+      // select TiDB component
+      await ppExpect(searchForm).toClick('div:nth-child(4)')
+      await ppExpect(page).toClick('ul[role=tree] span[title="TiDB"]')
```
Comment (breeswish) [inline ui/tests/e2e/search_log.test.ts]: give a id?

File: ui/tests/e2e/search_log.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    'should search correct logs',
+    async () => {
+      const page = await browser.newPage()
+
+      // login
+      await page.goto(LOGIN_URL)
+      await ppExpect(page).toClick('button#signin_btn')
+
+      // jump to search logs page
+      await ppExpect(page).toClick('a[href="#/search_logs"]')
```
Comment (breeswish) [inline ui/tests/e2e/search_log.test.ts]: prefer not to match by text or id

Comment (baurine) [reply ui/tests/e2e/search_log.test.ts]: I guess you mean prefer to match by text or id, right? fixed by that.

File: ui/tests/e2e/search_log.test.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      // find search form
+      const searchForm = await page.waitForSelector('form#search_form')
+
+      // set log level to INFO
+      await ppExpect(searchForm).toClick('div#log_level_selector')
+      await ppExpect(page).toClick('li[data-e2e="level_info"]')
+
+      // select TiDB component
+      await ppExpect(searchForm).toClick('div[data-e2e="components_selector"]')
+      await ppExpect(page).toClick('ul[role=tree] span[title="TiDB"]')
```
Comment (baurine) [inline ui/tests/e2e/search_log.test.ts]: The components selector is a dropdown with absolute position, its parent node is body, failed to add id or data-e2e to it.

File: ui/tests/e2e/search_log.test.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+      // start search
+      await ppExpect(searchForm).toClick('button#search_btn')
+
+      // check search result
+      const logsTable = await page.waitForSelector(
+        'div#logs_result table tbody'
```
Comment (breeswish) [inline ui/tests/e2e/search_log.test.ts]: `#logs_result` is fine

File: ui/tests/e2e/search_log.test.ts (hunk 3 above)
Comment (breeswish) [reply ui/tests/e2e/search_log.test.ts]: prefer to add comments in the test case

File: ui/tests/e2e/search_log.test.ts (hunk 4 above)
Comment (baurine) [reply ui/tests/e2e/search_log.test.ts]: yep, will do it soon.

File: ui/tests/e2e/search_log.test.ts (hunk 3 above)
Comment (baurine) [reply ui/tests/e2e/search_log.test.ts]: will do it soon.

**Pull request merged**